src/trackedit/internal/au3/au3interaction.cpp

```diff
@@ -1303,6 +1305,82 @@ void Au3Interaction::moveTracksTo(const TrackIdList& trackIds, int to)
     projectHistory()->pushHistoryState("Move track", "Move track");
 }
 
+void Au3Interaction::insertSilence(const TrackIdList trackIds, secs_t begin, secs_t end, secs_t duration)
+{
+    auto prj = globalContext()->currentTrackeditProject();
+    auto& project = projectRef();
+
+    if (trackIds.empty()) {
+        auto& tracks = Au3TrackList::Get(project);
+        auto& trackFactory = ::WaveTrackFactory::Get(project);
+
+        sampleFormat defaultFormat = QualitySettings::SampleFormatChoice();
+        auto rate = ::ProjectRate::Get(project).GetRate();
+
+        auto track = trackFactory.Create(defaultFormat, rate);
+        track->SetName(tracks.MakeUniqueTrackName(Au3WaveTrack::GetDefaultAudioTrackNamePreference()));
+
+        track->InsertSilence(begin, duration);
+
+        tracks.Add(track);
+
+        trackedit::ITrackeditProjectPtr prj = globalContext()->currentTrackeditProject();
+        prj->notifyAboutTrackAdded(DomConverter::track(track.get()));
+        projectHistory()->pushHistoryState(muse::trc("trackedit", "Insert silence"), muse::trc("trackedit", "Insert silence"));
+        return;
+    }
+
+    for (const auto& trackId : trackIds) {
+        Au3WaveTrack* waveTrack = DomAccessor::findWaveTrack(project, Au3TrackId(trackId));
```

Comment:
I think you do want to verify the validity of this pointer before proceeding: I've seen this return `nullptr` unexpectedly.

Reply:
That's some sloppy code, thanks!